contentpal/src/main/java/org/dmfs/android/contentpal/predicates/EqArg.java

```diff
@@ -42,16 +44,16 @@ public EqArg(@NonNull String columnName, @NonNull Object argument)
 
     @NonNull
     @Override
-    public CharSequence selection()
+    public CharSequence selection(@NonNull TransactionContext transactionContext)
     {
         return new StringBuilder(mColumnName).append(" = ?");
     }
 
 
     @NonNull
     @Override
-    public Iterable<String> arguments()
+    public Iterable<Argument> arguments(@NonNull TransactionContext transactionContext)
     {
-        return new SingletonIterable<>(mArgument.toString());
+        return new SingletonIterable<>((Argument) new ValueArgument(mArgument));
```

Comment:
Cast can be removed with specifying the generic parameter.

Reply:
right, fixed it